pkg/cloudprovider/providers/openstack/openstack_loadbalancer.go

```diff
@@ -1329,16 +1442,19 @@ func (lb *LbaasV1) EnsureLoadBalancer(clusterName string, apiService *v1.Service
 		}
 	}
 
-	lbmethod := pools.LBMethod(lb.opts.LBMethod)
-	if lbmethod == "" {
-		lbmethod = pools.LBMethodRoundRobin
+	// if this service has an annotation for LB method, use that instead
+	lbMethod := pools.LBMethod(getStringFromServiceAnnotation(apiService, ServiceAnnotationLoadBalancerLBMethod, lb.opts.LBMethod, lb.opts.EnableAnnotationLBMethod))
+
+	if lbMethod == "" {
+		lbMethod = pools.LBMethodRoundRobin
 	}
 	name := cloudprovider.GetLoadBalancerName(apiService)
+	subnetID := getStringFromServiceAnnotation(apiService, ServiceAnnotationLoadBalancerSubnetID, lb.opts.SubnetId, lb.opts.EnableAnnotationSubnetId)
```

Comment:
currently there are three ways to get subnet-id:
1.lbaas.opts.SubnetId
2.ServiceAnnotationLoadBalancerSubnetID
3.autoprobe subnet-id by "getSubnetIDForLB()"
You should also consider 3. If 1 and 2 are empty, we can use 3.